### PR TITLE
Feature 25/add login endpoint

### PR DESCRIPTION
**Pull request by Jeyob**
Massive commit, I know.

Adds the authentication endpoints to generate and verify tokens issued. This can later be replaced by an external authentication service (if we want to)

Effectivly introduces `../auth/register` and `../auth/login` endpoints which gives you back the token that we will later verify in each request to protected endpoints

### CODE REVIEW

File: app/auth_0_1/routes.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+from .. import db
+
+
+@auth.route('/register', methods=['POST'])
+def register_account():
+    if request.is_json:
+        account = Account.query.filter_by(email=request.json.get('email')).first()
+        # check that account does not exist
+        if account:
+            raise Conflict(error_message="account already exists")
```
Comment (doohoo) [inline app/auth_0_1/routes.py:18]: Maybe I am remembering things wrong here, but didn't we also say that "non-verified emails" should be able to register "again" (if they lost the email or the token expires)?

Comment (Jeyob) [reply app/auth_0_1/routes.py:18]: ffs, I knew I forgot somthing! 

Yepp, you're right. I think the solution to that was to check for whether the user existed but hadn't `confirmed` their account, in which case we just send the link again. 

will update

**Comment by Jeyob** [comment]
Updated, sorry for teh delay. Its been a busy week.

So this should align more to what we agreed on i.e.

When the user registers again, we check that it hasn't been verified and send a new email. If veriifed throw exception.

BONUS: also included an endpoint to use in order to get the status of the user. This can be useful as a way to check that the token is still valid or just get the basic information about the logged in user. Also, this is the first `PROTECTED ENDPOINT` which means that it will check that the `token` is valid first, then return the requested resource.

@doohoo

File: app/auth_0_1/routes.py (hunk 1 above)
Comment (Jeyob) [reply app/auth_0_1/routes.py]: PTAL 😄

**Pull request merged**